model/pipeline/src/main/proto/beam_runner_api.proto

```diff
@@ -1093,6 +1093,72 @@ message SideInput {
   FunctionSpec window_mapping_fn = 3;
 }
 
+message StandardArtifacts {
+  enum Types {
+    // A URN for artifacts stored in a local directory.
+    // payload: ArtifactFilePayload.
+    FILE     = 0 [(beam_urn) = "beam:artifact:file:v1"];
+
+    // A URN for artifacts embedded in ArtifactInformation proto.
+    // payload: EmbeddedFilePayload.
+    EMBEDDED = 1 [(beam_urn) = "beam:artifact:embedded:v1"];
+
+    // A URN for artifacts described by remote URLs.
+    // payload: a string for an artifact URL e.g. "https://.../foo.jar"
+    REMOTE     = 2 [(beam_urn) = "beam:artifact:remote:v1"];
+
+    // A URN for Python artifacts hosted on PYPI.
+    // payload: PypiPayload
+    PYPI     = 3 [(beam_urn) = "beam:artifact:pypi:v1"];
+
+    // A URN for Java artifacts hosted on Maven central.
+    // payload: MavenPayload
+    MAVEN    = 4 [(beam_urn) = "beam:artifact:maven:v1"];
+  }
+}
+
+message ArtifactFilePayload {
+  // A path to an artifact file on a local system.
+  string local_path = 1;
+
+  // A generated staged name (no path).
+  string staged_name = 2;
+}
+
+message EmbeddedFilePayload {
+  // raw data bytes for an embedded artifact
+  bytes data = 1;
+
+  // A generated staged name (no path).
+  string staged_name = 2;
+}
+
+message PypiPayload {
+  // Pypi compatible artifact id e.g. "apache-beam"
+  string pypi_artifact_id = 1;
+
+  // Pypi compatible version string.
+  string pypi_version_range = 2;
+}
+
+message MavenPayload {
+  // Maven compatible group id e.g. "org.apache.beam"
+  string maven_group_id = 1;
```

Comment:
Note that group/artifact/version isn't enough to fully specify all types of maven artifacts.

Either use a single string to specify the artifact
`groupId:artifactId:version[:packaging[:classifier]]`
where packaging and classifier are optional and packaging by default is "jar" and classifier is omitted if unspecified.

Or add `classifier` and `type` and rename `maven_group_id` to `group_id`, `maven_artifact_id` to `artifact_id` and `maven_version_range` to `version`.

Reply:
If there's a standard string format, let's just use that rather than breaking this up.

Reply:
Merged into a single field `artifact_specifier` which expects the format of `groupId:artifactId:version[:packaging[:classifier]]`